mungegithub/mungers/submit-queue.go

```diff
@@ -866,22 +866,34 @@ func (sq *SubmitQueue) validForMerge(obj *github.MungeObject) bool {
 	}
 
 	// Clearly
-	if !obj.HasLabel(lgtmLabel) {
+	if !(obj.HasLabel(lgtmLabel) || obj.HasLabel(approvedLabel)) {
 		sq.SetMergeStatus(obj, noLGTM)
 		return false
 	}
 
 	// PR cannot change since LGTM was added
 	lastModifiedTime := obj.LastModifiedTime()
 	lgtmTime := obj.LabelTime(lgtmLabel)
+	approvedTime := obj.LabelTime(approvedLabel)
 
-	if lastModifiedTime == nil || lgtmTime == nil {
+	if lastModifiedTime == nil || (lgtmTime == nil && approvedTime == nil) {
 		glog.Errorf("PR %d was unable to determine when LGTM was added or when last modified", *obj.Issue.Number)
 		sq.SetMergeStatus(obj, unknown)
 		return false
 	}
 
-	if lastModifiedTime.After(*lgtmTime) {
+	var timeToCompare *time.Time
+	if lgtmTime == nil {
+		timeToCompare = approvedTime
```

Comment:
What if the approvedTime is nil here? It wouldn't enter the `else if...`. Maybe the next condition shouldn't be `else if..`.